Tests/GitHubCodespaces.tests.ps1

```diff
@@ -0,0 +1,450 @@
+        BeforeAll {
+            $codespaces = Get-GitHubCodespace -OrganizationName $script:organizationName
+            $userCodespaces = Get-GitHubCodespace -OrganizationName $script:organizationName -UserName $script:OwnerName
+        }
+
+        It 'Should have results for the organization user' {
+            $userCodespaces.name | Should -BeIn $codespaces.name
+        }
+
+        It 'Should return the correct properties' {
+            foreach ($codespace in $userCodespaces)
+            {
+                $codespace.owner.login | Should -Be $script:OwnerName
+            }
+        }
+    }
+
+    Context 'When getting a codespace for a specified codespace name' {
+        BeforeAll {
+            $codespaces = Get-GitHubCodespace
+            $codespace = Get-GitHubCodespace -CodespaceName $codespaces[0].name
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespace.PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return the correct properties' {
+            $codespace.owner.login | Should -Be $script:OwnerName
+        }
+    }
+
+    Context 'When specifiying the Uri parameter' {
+        BeforeAll {
+            $codespace = Get-GitHubCodespace -Uri $repo.RepositoryUrl
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespace.PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return the correct properties' {
+            $codespace.owner.login | Should -Be $script:OwnerName
+            $codespace.repository.name | Should -Be $repo.name
+        }
+    }
+
+    Context "When specifiying the Uri parameter from the pipeline" {
+        BeforeAll {
+            $codespace = $repo | Get-GitHubCodespace
+        }
+
+        It 'Should return objects of the correct type' {
+            $codespace.PSObject.TypeNames[0] | Should -Be 'GitHub.Codespace'
+        }
+
+        It 'Should return the correct properties' {
+            $codespace.repository.name | Should -Be $repo.name
+        }
+    }
+}
+
+
+Describe 'GitHubCodespaces\New-GitHubCodespace' {
+
+    Context -Name 'When creating a repository for the authenticated user' {
+
+        Context -Name 'When creating a codespace with default settings with RepositoryId' {
+            BeforeAll {
+                $newGitHubCodespaceParms = @{
+                    RepositoryId = $repo.Id
+                }
+                $codespace = New-GitHubCodespace @newGitHubCodespaceParms
+                Start-Sleep -Seconds 2
+            }
+
+            It 'Should return an object of the correct type' {
+                $codespace | Should -BeOfType PSCustomObject
+            }
+
+            It 'Should return the correct properties' {
+                $codespace.display_name | Should -Not -BeNullOrEmpty
+                $codespace.repository.name | Should -Be $repo.name
+                $codespace.idle_timeout_minutes | Should -Be 30
```

Comment:
I didn't see `30` specified anywhere in the docs as the default.  Not sure it makes sense to explicitly test for this value when the default isn't defined and you didn't specify an expected value.

Reply:
Valid.  The default can be changed by an admin.  Removing these 30-minute assertions.